### PR TITLE
[Profiler] Update third_party/kineto submodule hash

### PR DESCRIPTION
**Pull request by aaronenyeshi**
Summary:
Include improvements such as:
- AMD: roctracer crash fix and roctracer external correlations
- NCCL metadata: process group id to process group name
- Complete nanosecond transition for Kineto
- Remove PrivateUse1Type function causing gpu track to be above cpu tracks
- Use relative time and fix gpu user annotation causing events to overlap

Test Plan: CI and Github CI (full suite)

Reviewed By: sraikund16

Differential Revision: D56475055

Pulled By: aaronenyeshi

### CODE REVIEW

**Comment by aaronenyeshi** [comment]
cc @@fwenguang

**Review by davidberard98 — APPROVED** [review]
LGTM (pending CI)!

**Review by malfet — APPROVED** [review]
LGTM, but please don't merge -f it especially if it will fail profiler tests

**Pull request closed without merging**
